deps/graph-builder/src/lockfileToDepGraph.ts

```diff
@@ -183,6 +183,19 @@ export async function lockfileToDepGraph (
             throw err
           }
         }
+        const pkgNameAndVersion = `${pkgName}@${pkgVersion}`
+        let patchInfo: PatchInfo | undefined
+        if (opts.patchedDependencies?.[pkgNameAndVersion]) {
+          patchInfo = {
+            allowFailure: false,
+            file: opts.patchedDependencies[pkgNameAndVersion],
+          }
+        } else if (opts.patchedDependencies?.[pkgName]) {
+          patchInfo = {
+            allowFailure: true,
```

Comment:
I'd rename this to something like `appliedToAnyVersion`.

Reply:
I'm not sure. The only purpose of this field is to skip failures, I don't know what other uses could the consumer code infer from `appliedToAnyVersion`.

Furthermore, what if we want to allow the user to decide whether to skip failure in the future? Would an additional field like `allowFailure` be added?

That being said, the final decision is yours. If you still insist on `appliedToAnyVersion`, then I shall oblige.

Reply:
You can keep the name of the variable in other places. But this package is not applying the patch, not sure why it needs to know about how the patch will be applied. All it knows is that the patch was declared for a single version or for all versions.

Reply:
@zkochan https://github.com/pnpm/pnpm/pull/8337/commits/466b52b427b93cad1f011a1eb820215872afd8cf